src/components/Header/HeaderNavigation.jsx

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import * as React from "react";
+
+import * as Rivet from "../util/Rivet";
+import Icon, { IconNames } from "../util/RivetIcons";
+import { useEffect, useRef } from "react";
+import { handler, isUnhandledKeyPress } from "./HeaderEventUtils";
+import {
+  isEscapeKeyPress,
+  isMouseEvent,
+  isTabKeyPress,
+  targets,
+} from "../util/EventUtils";
+import { TestUtils } from "../util/TestUtils";
+import { renderHeaderNavUnorderedList } from "./childUtils";
+import { findFirstChildOfType } from "../util/childUtils";
+import HeaderAvatar from "./HeaderAvatar";
+
+const HeaderNavigation = ({ children, ...attrs }) => {
+  const [isNavMenuOpen, setIsNavMenuOpen] = React.useState(false);
+
+  const wrapperDivRef = useRef(null);
+  const toggleButtonRef = useRef(null);
+
+  useEffect(() => {
+    handleEventRegistration();
+    return () => {
+      eventHandler.deregister();
+    };
+  });
+
+  const avatarChild = findFirstChildOfType(children, HeaderAvatar.displayName);
+
+  const toggleNavigation = () => {
+    setIsNavMenuOpen(!isNavMenuOpen);
+  };
+
+  const handleEvent = (event) => {
+    if (event && shouldToggleNavigation(event)) {
+      toggleNavigation(event);
+      if (isEscapeKeyPress(event)) {
+        toggleButtonRef.current.focus();
+      }
+    }
+  };
```

Comment:
I moved all the shouldToggle* methods outside the components. However, moving methods like handleEvent outside the component would increase the number of arguments that these methods receive, which IMO makes the code difficult to follow functionally.

Moving the shouldToggle* method outside was fine, since we are only adding one extra argument to those by moving them outside. There are methods like [handleArrowKeyPress](https://github.com/indiana-university/rivet-react/blob/feature/2-0-0-header/src/components/Header/HeaderMenu.jsx#L96) where we would be adding up to 4 new arguments, that don't  come together in a meaningful way inside the method's definition.

That's my concern with not going in this direction.